src/stream/src/executor/source/source_backfill_executor.rs

```diff
@@ -54,9 +54,15 @@ pub enum BackfillState {
     SourceCachingUp(String),
     Finished,
 }
-pub type BackfillStates = HashMap<SplitId, BackfillState>;
+pub type BackfillStates = HashMap<SplitId, BackfillStateWithCnt>;
 
-impl BackfillState {
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct BackfillStateWithCnt {
+    pub state: BackfillState,
+    pub num_consumed_rows: u64,
+}
+
+impl BackfillStateWithCnt {
     pub fn encode_to_json(self) -> JsonbVal {
         serde_json::to_value(self).unwrap().into()
     }
```

Comment:
This changes the state table representation, and should be a breaking change. But if we can get this into 2.1, then we don't need to consider compatibility.